### ISSUE TITLE
"target" : "package" resolves transitive dependencies as "project"

### ISSUE THREAD

**Issue by brthor** [NuGet/Home#2818]
Current repro is in the cli repo, I can put together an independent repro if necessary. 

Steps:
1. `git clone https://github.com/dotnet/cli`
2. `git checkout brthor/nuget-transitive-project-bug`
3. Acquire latest CLI: https://dotnetcli.blob.core.windows.net/dotnet/preview/Installers/Latest/dotnet-dev-win-x64.latest.exe
4. `cd %cliRepoRoot%/tools/RuntimeGraphGenerator`
5. Notice global.json + project.json with `target: package` in this directory
6. `dotnet restore`
7. Find below in lock file:

```
"Microsoft.DotNet.InternalAbstractions/1.0.0-rc3": {
        "type": "project",
        "framework": ".NETStandard,Version=v1.3",
        "dependencies": {
          "System.AppContext": "4.1.0-rc2-24027",
          "System.Collections": "4.0.11-rc2-24027",
          "System.IO": "4.1.0-rc2-24027",
          "System.IO.FileSystem": "4.0.1-rc2-24027",
          "System.Reflection.TypeExtensions": "4.1.0-rc2-24027",
          "System.Runtime.Extensions": "4.1.0-rc2-24027",
          "System.Runtime.InteropServices": "4.1.0-rc2-24027",
          "System.Runtime.InteropServices.RuntimeInformation": "4.0.0-rc2-24027"
        },
        "compile": {
          "netstandard1.3/Microsoft.DotNet.InternalAbstractions.dll": {}
        },
        "runtime": {
          "netstandard1.3/Microsoft.DotNet.InternalAbstractions.dll": {}
        }
      },
```

The transitive dependencies of the "target": "package" are being resolved as projects, and the global.json in the immediate folder is being ignored.

**Comment by brthor** [comment]
cc @emgarten @eerhardt

**Comment by brthor** [comment]
I managed to get this working by moving the global.json one directory up, but even needing that is a workaround when `"target": "package"` is specified.

**Comment by emgarten** [comment]
Target package just requires that top level dependency to be of that type, not all children. If target: project is specified it is still expected that the project can have child package dependencies.

Global.json needs to be in a parent directory.

**Comment by rrelyea** [comment]
thanks for raising the issues. helps us understand potentially confusing issues.
We believe in the the project/package search, and requiring override at each dependency.
Global.json and nuget.config are likely consolidated in future, so not going to invest in global.json happiness too much now.

**Comment by brthor** [comment]
That explanation makes sense, but I'm still somewhat confused. Please offer some help on this.

Why do the project dependencies win over package dependencies for transitive dependencies, when the exact version of the transitive dependencies exists on the feed, and the project dependency is not an exact version match?

I'm confused because the project dependency version resolves to `1.0.0-rc3` (in the lock file), and the nuspec of the `target: package`, package, contains `<dependency id="Microsoft.Extensions.DependencyModel" version="[1.0.0-rc3-002771, )" />` yet the dependency still resolves to the project.

Is `1.0.0-rc3` considered higher or lower than `1.0.0-rc3-002771`? Are project dependencies given precedence?

**Comment by emgarten** [comment]
The dependency version isn't used when selecting the project that best matches the dependency, just the id, and a project with a matching id always wins over a package. Without this it would be easy to end up pulling in an old package instead of a project right in your solution, in your scenario you are trying to use the package instead, but this is a rare scenario.
